lib/Jsrt/ChakraCore.h

```diff
@@ -130,6 +130,22 @@ typedef struct JsNativeFunctionInfo
 /// <returns>The result of the call, if any.</returns>
 typedef _Ret_maybenull_ JsValueRef(CHAKRA_CALLBACK * JsEnhancedNativeFunction)(_In_ JsValueRef callee, _In_ JsValueRef *arguments, _In_ unsigned short argumentCount, _In_ JsNativeFunctionInfo *info, _In_opt_ void *callbackState);
 
+/// <summary>
+///     A Promise Rejection Tracker callback.
+/// </summary>
+/// <remarks>
+///     The host can specify a promise rejection tracker callback in <c>JsSetHostPromiseRejetionTracker</c>.
+///     If a promise is rejected with no reactions or a reaction is added to a promise that was rejected
+///     before it had reactions by default nothing is done.
+///     A Promise Rejection Tracker callback may be set - which will then be called when this occurs.
+/// </remarks>
```

Comment:
Please add a note here stating that this function must never throw any exceptions, as required by the spec.

I'd also like if you could add a comment about the lifetimes of the arguments; I believe that the promise handed to the callback here may be collected at any time (especially if it is an unhandled rejection which is never handled, or if it is a newly-handled rejection which will shortly fall out of scope and become unreferenced) and there are no particular steps taken to pin either of the arguments for this function.

Reply:
I believe the specification includes wording implying the host is responsible for managing lifetime of the promise, but agreed it should be documented here too, for visibility if nothing else.

Reply:
I note in particular that the spec specifically forbids the host from holding onto the promise after a “handled” event.

Reply:
Updated comments and re-pushed